### PR TITLE
Minor documentation correction for 2.14.0

### PR DESCRIPTION
**Pull request by reta**
### Description
Minor documentation correction for 2.14.0, incorporate changes from:
- https://github.com/opensearch-project/OpenSearch/pull/13532
- https://github.com/opensearch-project/OpenSearch/pull/13174

### Issues Resolved
N/A

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@hdhalter a few correction for 2.14 please, thank you!

File: _field-types/supported-field-types/date.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ OpenSearch has built-in date formats, but you can also create your own custom fo
 
 ## Default format
 
-As of OpenSearch 2.12, the default date format is `strict_date_time_no_millis||strict_date_optional_time||epoch_millis`. To revert the default format back to `strict_date_optional_time||epoch_millis` (the default format for OpenSearch 2.11 and earlier), set the `opensearch.experimental.optimization.datetime_formatter_caching.enabled` feature flag to `false`. For more information about enabling and disabling feature flags, see [Enabling experimental features]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/experimental/).
+As of OpenSearch 2.12, there is a new experimental default date format `strict_date_time_no_millis||strict_date_optional_time||epoch_millis` (the default format is `strict_date_optional_time||epoch_millis`). To use it, please set the `opensearch.experimental.optimization.datetime_formatter_caching.enabled` feature flag to `true`. For more information about enabling and disabling feature flags, see [Enabling experimental features]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/experimental/).
```
Comment (Naarcha-AWS) [inline _field-types/supported-field-types/date.md:65]: A little confused here @reta. Are you saying that as of OpenSearch 2.12, the default has changed from `strict_date_time_no_millis` to `strict_date_optional_time||epoch_millis` as long as the experimental setting is enabled? 

```suggestion
As of OpenSearch 2.12, there is a new experimental default date format `strict_date_time_no_millis||strict_date_optional_time||epoch_millis` (the default format is `strict_date_optional_time||epoch_millis`). To use it,  set the `opensearch.experimental.optimization.datetime_formatter_caching.enabled` feature flag to `true`. For more information about enabling and disabling feature flags, see [Enabling experimental features]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/experimental/).
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: _field-types/supported-field-types/date.md (hunk 1 above)
Comment (reta) [reply _field-types/supported-field-types/date.md:65]: > A little confused here @reta. Are you saying that as of OpenSearch 2.12, the default has changed from `strict_date_time_no_millis` to `strict_date_optional_time||epoch_millis` as long as the experimental setting is enabled?

Thanks @Naarcha-AWS , what happened is that we rolled out experimental settings that was set to `true` by default and changed the default datetime format. It caused issues to our users so we are basically correcting that by deactivating this experimental  setting by default, so the default datetime format stays as it was before.

Comment (Naarcha-AWS) [reply _field-types/supported-field-types/date.md]: So the default format would be `strict_date_time_no_millis||strict_date_optional_time||epoch_millis` when `the `opensearch.experimental.optimization.datetime_formatter_caching.enabled` feature flag is set to `true`?

Comment (reta) [reply _field-types/supported-field-types/date.md]: > So the default format would be `strict_date_time_no_millis||strict_date_optional_time||epoch_millis` when the `opensearch.experimental.optimization.datetime_formatter_caching.enabled` feature flag is set to `true`?

That is correct

File: _field-types/supported-field-types/date.md [hunk 2]
```diff
@@ -62,7 +62,7 @@ OpenSearch has built-in date formats, but you can also create your own custom fo
 
 ## Default format
 
-As of OpenSearch 2.12, the default date format is `strict_date_time_no_millis||strict_date_optional_time||epoch_millis`. To revert the default format back to `strict_date_optional_time||epoch_millis` (the default format for OpenSearch 2.11 and earlier), set the `opensearch.experimental.optimization.datetime_formatter_caching.enabled` feature flag to `false`. For more information about enabling and disabling feature flags, see [Enabling experimental features]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/experimental/).
+As of OpenSearch 2.12, there is a new experimental date format `strict_date_time_no_millis||strict_date_optional_time||epoch_millis` (the default format is `strict_date_optional_time||epoch_millis`). To use it,  set the `opensearch.experimental.optimization.datetime_formatter_caching.enabled` feature flag to `true`. For more information about enabling and disabling feature flags, see [Enabling experimental features]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/experimental/).
```
Comment (Naarcha-AWS) [inline _field-types/supported-field-types/date.md:65]:
```suggestion
As of OpenSearch 2.12, you can to choose to use an experimental default date format, `strict_date_time_no_millis||strict_date_optional_time||epoch_millis`. To use the experimental default,  set the `opensearch.experimental.optimization.datetime_formatter_caching.enabled` feature flag to `true`. For more information about enabling and disabling feature flags, see [Enabling experimental features]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/experimental/).
```

File: _field-types/supported-field-types/date.md [hunk 3]
```diff
@@ -62,7 +62,7 @@ OpenSearch has built-in date formats, but you can also create your own custom fo
 
 ## Default format
 
-As of OpenSearch 2.12, the default date format is `strict_date_time_no_millis||strict_date_optional_time||epoch_millis`. To revert the default format back to `strict_date_optional_time||epoch_millis` (the default format for OpenSearch 2.11 and earlier), set the `opensearch.experimental.optimization.datetime_formatter_caching.enabled` feature flag to `false`. For more information about enabling and disabling feature flags, see [Enabling experimental features]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/experimental/).
+As of OpenSearch 2.12, you can to choose to use an experimental default date format, `strict_date_time_no_millis||strict_date_optional_time||epoch_millis`. To use the experimental default,  set the `opensearch.experimental.optimization.datetime_formatter_caching.enabled` feature flag to `true`. For more information about enabling and disabling feature flags, see [Enabling experimental features]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/experimental/).
```
Comment (Naarcha-AWS) [inline _field-types/supported-field-types/date.md:65]:
```suggestion
As of OpenSearch 2.12, you can to choose to use an experimental default date format, `strict_date_time_no_millis||strict_date_optional_time||epoch_millis`. To use the experimental default, set the `opensearch.experimental.optimization.datetime_formatter_caching.enabled` feature flag to `true`. For more information about enabling and disabling feature flags, see [Enabling experimental features]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/experimental/).
```

**Review by Naarcha-AWS — APPROVED** [review]

**Pull request merged**